src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolverOptionsSetup.cs

```diff
@@ -62,41 +93,192 @@ private string GetWorkersRootDirPath(IConfiguration configuration)
+                {
+                    probingPaths.Add(path);
+                }
+            }
+
+            return probingPaths;
+        }
+
+        /// <summary>
+        /// Returns the default site extensions path for Windows environment.
+        /// </summary>
+        private string GetWindowsSiteExtensionsPath()
+        {
+            var assemblyDir = AppContext.BaseDirectory;
+
+            //Move 2 directories up to get to the SiteExtensions directory. Example output: "c:\\home\\SiteExtensions"
+            return Directory.GetParent(assemblyDir.TrimEnd(Path.DirectorySeparatorChar))?.Parent?.FullName;
+        }
+
+        /// <summary>
+        /// Returns a set of worker runtimes available for dynamic resolution from hosting config.
+        /// </summary>
+        internal HashSet<string> GetWorkersAvailableForResolution() =>
+            (_functionsHostingConfigOptions.Value.WorkersAvailableForDynamicResolution ?? string.Empty)
+            .Split('|', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)
+            .ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Converts language workers related configuration sections to a dictionary.
+        /// </summary>
+        internal static Dictionary<string, string> GetLanguageWorkersSettings(IConfiguration configuration)
+        {
+            var languageWorkersSettings = new Dictionary<string, string>();
+
+            foreach (var kvp in configuration.AsEnumerable())
+            {
+                if (kvp.Key.StartsWith(RpcWorkerConstants.LanguageWorkersSectionName))
+                {
+                    languageWorkersSettings[kvp.Key] = kvp.Value;
                 }
             }
 
-            string configSectionValue = GetConfigurationSectionValue(_configuration, nameof(_configuration), requiredSection);
-            if (!string.IsNullOrEmpty(configSectionValue))
+            return languageWorkersSettings;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether dynamic worker resolution is enabled.
+        /// Users can disable dynamic worker resolution via setting the appropriate feature flag.
+        /// Worker resolution can be enabled for specific workers at the stamp level via hosting config options.
+        /// Feature flag takes precedence over hosting config options.
+        /// </summary>
+        internal bool IsDynamicWorkerResolutionEnabled(string workerRuntime, HashSet<string> workersAvailableForResolution, bool isPlaceholderModeEnabled, bool isMultiLanguageEnv)
+        {
+            if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableDynamicWorkerResolution, _environment) || workersAvailableForResolution.Count == 0)
             {
-                return _configuration;
+                return false;
             }
 
-            return null;
+            if (!isPlaceholderModeEnabled && !isMultiLanguageEnv && !string.IsNullOrWhiteSpace(workerRuntime))
```

Comment:
we don't want this to use the external workers in placeholder mode? or will this eventually be changed?

Reply:
External workers will be used in placeholder mode if they are available for resolution. If at least one worker is available for resolution, the feature will be enabled for placeholder mode and multi-language worker environment scenario.

If it is not a placeholder mode and multi-language env, the feature will be enabled only if the worker specified by workerRuntime is available for resolution.

Reply:
I did not fully understand this. This method currently returns `true` in placeholder mode without checking `workersAvailableForResolution`. Is that intentional? Shouldn’t we always check `workersAvailableForResolution` regardless of whether we are in placeholder or non-placeholder mode?